add ticdc best pratice and FAQ (#11641)
pingcap/docs: ticdc/ticdc-overview.md

As shown in the preceding architecture diagram, TiCDC supports replicating data to TiDB, MySQL, and Kafka databases.

## Best practices

> - When using TiCDC to replicate data between two TiDB clusters and the latency between them is greater than 100 ms, it is recommended that you deploy TiCDC in the region (IDC) where the downstream TiDB cluster is located.

Reviewer: ```suggestion
- When using TiCDC to replicate data between two TiDB clusters and the network latency between them is greater than 100 ms, it is recommended that you deploy TiCDC in the region (IDC) where the downstream TiDB cluster is located.
```

> - When using TiCDC to replicate data between two TiDB clusters and the network latency between them is greater than 100 ms, it is recommended that you deploy TiCDC in the region (IDC) where the downstream TiDB cluster is located.

Reviewer 2: ```suggestion
- When you use TiCDC to replicate data between two TiDB clusters and the network latency between the clusters is higher than 100 ms, it is recommended that you deploy TiCDC in the region (IDC) where the downstream TiDB cluster is located.
```

- TiCDC only replicates the table that has at least one **valid index**. A **valid index** is defined as follows:

    - A primary key (`PRIMARY KEY`) is a valid index.

>     - A unique index (`UNIQUE INDEX`) is valid if every column of it is explicitly defined as non-nullable (`NOT NULL`) and the index does not have the virtual generated column (`VIRTUAL GENERATED COLUMNS`).

Reviewer 2: ```suggestion
    - A unique index (`UNIQUE INDEX`) is valid if every column of the index is explicitly defined as non-nullable (`NOT NULL`) and the index does not have the virtual generated column (`VIRTUAL GENERATED COLUMNS`).
```

> - To use TiCDC in DR scenarios, you need to configure [redo log](/ticdc/ticdc-sink-to-mysql.md#eventually-consistent-replication-in-disaster-scenarios).

Reviewer 2: ```suggestion
- To use TiCDC in disaster recovery scenarios, you need to configure [redo log](/ticdc/ticdc-sink-to-mysql.md#eventually-consistent-replication-in-disaster-scenarios).
```

> - When replicating a wide table with a large single row (greater than 1K), it is recommended that you configure [per-table-memory-quota](/ticdc/ticdc-server-config.md) so that `per-table-memory-quota` = `ticdcTotalMemory`/(`tableCount` * 2). `ticdcTotalMemory` is the memory of a TiCDC node, and `tableCount` is the number of target tables that a TiCDC node replicates.

Reviewer 2: ```suggestion
- When you replicate a wide table with a large single row (greater than 1K), it is recommended that you configure [`per-table-memory-quota`](/ticdc/ticdc-server-config.md) so that `per-table-memory-quota` = `ticdcTotalMemory`/(`tableCount` * 2). `ticdcTotalMemory` is the memory of a TiCDC node, and `tableCount` is the number of target tables that a TiCDC node replicates.
```

> **Note:**
>
> Since v4.0.8, TiCDC supports replicating tables **without a valid index** by modifying the task configuration. However, this compromises the guarantee of data consistency to some extent. For more details, see [Replicate tables without a valid index](/ticdc/ticdc-manage-changefeed.md#replicate-tables-without-a-valid-index).

## Restrictions

> 

Reviewer 2: Line 89 to Line 98 should be deleted according the docs-cn PR?
